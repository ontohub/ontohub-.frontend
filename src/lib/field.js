import React from "react";
import { Form, Label } from "semantic-ui-react";

export const Field = ({
  bar,
  errors,
  fieldId,
  label,
  name,
  onBlur,
  onChange,
  touched,
  type,
  value
}) => (
  <Form.Field>
    <label htmlFor={fieldId}>{label}</label>
    <input
      id={fieldId}
      name={name}
      onBlur={onBlur}
      onChange={onChange}
      placeholder={label}
      type={type}
      value={value}
    />
    {bar || null}
    {!!touched &&
      !!errors && (
        <Label pointing color="red">
          {errors.map(e => <div key={e}>{e}</div>)}
        </Label>
      )}
  </Form.Field>
);
