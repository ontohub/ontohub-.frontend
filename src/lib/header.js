import React from "react";
import styled from "styled-components";

const Header = props => (
  <StyledHeader>
    <InnerHeader
      heightTransitionDuration={props.heightTransitionDuration}
      contentTransitionDuration={props.contentTransitionDuration}
    >
      {props.children}
    </InnerHeader>
  </StyledHeader>
);

const StyledHeader = styled.div`
  background-color: ${({ theme }) => theme.colors.dark};
  color: ${({ theme }) => theme.colors.light} !important;
  margin-bottom: 1em;
  margin-top: -100vh;
  padding-top: 100vh;
`;

const InnerHeader = styled.div`
  max-width: ${({ theme }) => theme.sizes.contentWidth};
  margin: auto;
  padding-top: ${({ theme }) => theme.sizes.headerPadding};
`;

export { Header };
