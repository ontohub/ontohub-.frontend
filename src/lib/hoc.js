import React from "react";

const HOC = ({ hoc = o => o, component, ...props }) =>
  React.createElement(hoc(component), props);

export { HOC };
