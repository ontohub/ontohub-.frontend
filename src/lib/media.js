import { css } from "styled-components";

// These sizes and key names are tailored to Semantic-UI
const sizes = {
  largeScreenOnly: { min: 1200 },
  computerOnly: { min: 992, max: 1199 },
  tabletOnly: { min: 768, max: 991 },
  mobileOnly: { max: 767 },

  largeScreen: { min: 1200 },
  computer: { min: 992 },
  tablet: { min: 768 },
  mobile: { min: 0 }
};

// use em in breakpoints to work properly cross-browser and support users
// changing their browsers font-size: https://zellwk.com/blog/media-query-units/
const emSize = pxSize => pxSize / 16;

// iterate through the sizes and create a media template
export const media = Object.keys(sizes).reduce((accumulator, label) => {
  const intervals = [];
  if (sizes[label].min) {
    intervals.push(`(min-width: ${emSize(sizes[label].min)}em)`);
  }
  if (sizes[label].max) {
    intervals.push(`(max-width: ${emSize(sizes[label].max)}em)`);
  }
  accumulator[label] = (...args) => css`
    @media ${intervals.join(" and ")} {
      ${css(...args)};
    }
  `;
  return accumulator;
}, {});
