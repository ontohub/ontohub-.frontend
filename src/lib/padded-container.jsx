import React from "react";
import styled from "styled-components";
import { Container } from "semantic-ui-react";

export const PaddedContainer = styled(Container)`
  padding-bottom: ${({ theme }) => theme.sizes.headerPadding};
`;
