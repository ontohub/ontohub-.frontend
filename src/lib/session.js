const signOut = client => {
  localStorage.removeItem("auth-token");
  return client.resetStore();
};

const signIn = (client, token) => {
  if (token) {
    localStorage.setItem("auth-token", token);
    client.resetStore();
  }
};

export { signIn, signOut };
