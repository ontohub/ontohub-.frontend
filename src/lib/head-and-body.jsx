import React, { Fragment } from "react";
import { HOC } from "lib/hoc";
import { Header } from "lib/header";

const HeadAndBody = ({ hoc, head: Head, body: Body, ...props }) => (
  <HOC
    hoc={hoc}
    component={props => (
      <Fragment>
        <Header>
          <Head {...props} />
        </Header>
        <Body {...props} />
      </Fragment>
    )}
    {...props}
  />
);

export { HeadAndBody };
