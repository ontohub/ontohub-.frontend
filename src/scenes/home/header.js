import React from "react";
import { Header as SemHeader, List } from "semantic-ui-react";
import { PaddedContainer } from "lib/padded-container";
import { Item } from "./item";

export const Header = () => (
  <PaddedContainer>
    <SemHeader as="h2" inverted>
      Welcome to Ontohub
    </SemHeader>
    <List size="large">
      <Item icon="fork" header="Open" content="based on open source software" />
      <Item
        icon="code"
        header="Flexible"
        content="supporting OWL, UML, FOL/TPTP, HOL/THF, and more"
      />
      <Item
        icon="sitemap"
        header="Distributed"
        content="OMS alignments, mappings, networks, combinations using DOL"
      />
    </List>
  </PaddedContainer>
);
