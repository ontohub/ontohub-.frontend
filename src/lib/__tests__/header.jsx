import React from "react";
import { Header } from "lib/header";
import { ThemeProvider } from "styled-components";

describe("Header", () => {
  it("matches the snapshot", () => {
    const wrapper = render(
      <ThemeProvider
        theme={{ colors: { dark: "#000" }, sizes: { contentWidth: "700px" } }}
      >
        <Header>Header content</Header>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
