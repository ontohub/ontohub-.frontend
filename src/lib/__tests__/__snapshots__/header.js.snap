// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c0 {
  background-color: #000;
  color: !important;
  margin-bottom: 1em;
  margin-top: -100vh;
  padding-top: 100vh;
}

.c1 {
  max-width: 700px;
  margin: auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Header content
  </div>
</div>
`;
